src/client/js/utils/url.js

```diff
@@ -0,0 +1,66 @@
+
+const STORAGE_DOMAIN = "wr.io";
+
+export function extractFileName(pathname) {
+    var fileName = pathname.match(/\/[0-9]+\/(.*)/);
+    var out;
+    if (fileName) {
+        out = fileName[1];
+        if (out === "" || !out) {
+            out = "index.html"; // if no file specified, let's assume this is index.htm
```

Comment:
We have to store as index.html always. The name of file is for dir. So we get wr.io/ID/file_name/index.html